bindgen/ir/var.rs

```diff
@@ -389,9 +392,94 @@ impl ClangSubItemParser for Var {
     }
 }
 
+fn parse_macro_clang_fallback(
+    ctx: &mut BindgenContext,
+    cursor: &clang::Cursor,
+) -> Option<(Vec<u8>, cexpr::expr::EvalResult)> {
+    let file = ".macro_eval.c".to_string();
+    let contents = format!("int main() {{ {}; }}", cursor.spelling(),);
+
+    let root = if let Some(ftu) = ctx.fallback_translation_unit_mut() {
+        ftu.reparse(&[clang::UnsavedFile::new(&file, &contents)]);
+        ftu.translation_unit().cursor().collect_children()
+    } else {
+        let index = clang::Index::new(false, false);
+
+        let mut c_args = Vec::new();
+        for input_header in ctx.options().input_headers.iter() {
+            let path = Path::new(input_header.as_ref());
+            let header_name = path
+                .file_name()
+                .and_then(|hn| hn.to_str())
+                .map(|s| s.to_owned());
+            let header_path = path
+                .parent()
+                .and_then(|hp| hp.to_str())
+                .map(|s| s.to_owned());
+
+            if let (Some(hp), Some(hn)) = (header_path, header_name) {
```

Comment:
How would you recommend configuring this? Environment variables? Is the same directory okay as a default?

So one note is that the temporary `.c` file is created in the current working directory. Now that may not be a problem, but I'm wondering if it'll be subject to the same issue you mentioned above. Maybe we should allow a build directory that's configurable. We could theoretically put all of the temporary files in there for example.

Reply:
> How would you recommend configuring this? Environment variables?

I think a flag would be fine, e.g. `rustc` has `--out-dir`, `-Ztemps-dir`, `- Zdump-mir`; GCC has `-save-temps`, `-dumpdir`...

> Is the same directory okay as a default?

I guess, yeah -- `rustc` also has a similar default (the current working directory) so I guess there is "prior art" in that sense :) Though it is a bit of a pitfall.

Reply:
Also just a follow up here, would you recommend returning `None` if the path isn't valid instead of skipping the header so we don't fail silently for example?